tests/Integration/Behaviour/Features/Scenario/Product/update_stock_advanced.feature

```diff
@@ -0,0 +1,292 @@
+      | use_advanced_stock_management | true                 |
+    When I update product "productPack1" stock with following information:
+      | pack_stock_type               | products_only |
+    Then product "productPack1" should have following stock information:
+      | pack_stock_type | products_only |
+    # Of course stock type on both works as well
+    When I update product "productPack1" stock with following information:
+      | pack_stock_type               | both |
+    Then product "productPack1" should have following stock information:
+      | pack_stock_type | both |
+    # We can even switch back to default configuration
+    When I update product "productPack1" stock with following information:
+      | pack_stock_type               | default |
+    Then product "productPack1" should have following stock information:
+      | pack_stock_type | default |
+
+  Scenario: I update product out of stock
+    Given I add product "product1" with following information:
+      | name       | en-US:Presta camera |
+      | is_virtual | false               |
+    And product "product1" should have following stock information:
+      | out_of_stock_type | default |
+    When I update product "product1" stock with following information:
+      | out_of_stock_type | available |
+    Then product "product1" should have following stock information:
+      | out_of_stock_type | available |
+    When I update product "product1" stock with following information:
+      | out_of_stock_type | not_available |
+    Then product "product1" should have following stock information:
+      | out_of_stock_type | not_available |
+    When I update product "product1" stock with following information:
+      | out_of_stock_type | default |
+    Then product "product1" should have following stock information:
+      | out_of_stock_type | default |
+    When I update product "product1" stock with following information:
+      | out_of_stock_type | invalid |
+    Then I should get error that out of stock type is invalid
+
+  Scenario: Virtual product is available out of stock by default
+    Given I add product "product1" with following information:
+      | name       | en-US:eBook |
+      | is_virtual | true        |
+    Then product "product1" should have following stock information:
+      | out_of_stock_type | available |
+
+  Scenario: I update product quantity
+    Given I add product "product1" with following information:
+      | name       | en-US:Presta camera |
+      | is_virtual | false               |
+    And product "product1" should have following stock information:
+      | quantity | 0 |
+    When I update product "product1" stock with following information:
+      | quantity | 51 |
+    Then product "product1" should have following stock information:
+      | quantity | 51 |
+    And product "product1" last stock movement has following details:
+      | physical_quantity | 51 |
+      | sign              | 1  |
```

Comment:
what `sign` stands for? (I know its data column of stock movement), but what does that word mean?

Reply:
It's used in StockMovement to indicate if the stock was increased (`1`) or decreased (`-1`) It would be more accurate to use `+` or `-` but I went for the simplicity here and just checked the actual value in DB
Do you think I should add some transformations in the Behat context so that we use `+-` in the feature file?

Reply:
yeah sign is very confusing, I would go for `+-`, its much clearer.